client/src/main/java/io/pravega/client/control/impl/ControllerImpl.java

```diff
@@ -1306,6 +1368,15 @@ public CompletableFuture<PravegaNodeUri> getEndpointForSegment(final String qual
                 });
     }
 
+    @VisibleForTesting
+    public CompletableFuture<PravegaNodeUri> getPravegaNodeUriForTesting(String qualifiedSegmentName) {
```

Comment:
Instead of creating a new wrapping method, change the existing method package scope, put the test in the same Java package (client/controller/impl), and then add the annotation to it directly.

Reply:
Incorporated changes